gradio/components/file_explorer.py

```diff
@@ -156,80 +155,40 @@ def postprocess(self, value: str | list[str] | None) -> FileExplorerData | None:
+                continue
+            if self.ignore_glob and fnmatch.fnmatch(full_path, self.ignore_glob):
+                continue
+            target = files if is_file else folders
+            target.append(
+                {
+                    "name": item,
+                    "type": "file" if is_file else "folder",
+                    "valid": valid_by_glob,
+                }
+            )
 
-        def expand_braces(text, seen=None):
-            if seen is None:
-                seen = set()
-
-            spans = [m.span() for m in re.finditer("{[^{}]*}", text)][::-1]
-            alts = [text[start + 1 : stop - 1].split(",") for start, stop in spans]
-
-            if len(spans) == 0:
-                if text not in seen:
-                    yield text
-                seen.add(text)
-
-            else:
-                for combo in itertools.product(*alts):
-                    replaced = list(text)
-                    for (start, stop), replacement in zip(spans, combo):
-                        replaced[start:stop] = replacement
-
-                    yield from expand_braces("".join(replaced), seen)
-
-        def make_tree(files):
-            tree = []
-            for file in files:
-                parts = file.split(os.path.sep)
-                make_node(parts, tree)
-            return tree
-
-        def make_node(parts, tree):
-            _tree = tree
-            for i in range(len(parts)):
-                if _tree is None:
-                    continue
-                if i == len(parts) - 1:
-                    type = "file"
-                    _tree.append({"path": parts[i], "type": type, "children": None})
-                    continue
-                type = "folder"
-                j = next(
-                    (index for (index, v) in enumerate(_tree) if v["path"] == parts[i]),
-                    None,
-                )
-                if j is not None:
-                    _tree = _tree[j]["children"]
-                else:
-                    _tree.append({"path": parts[i], "type": type, "children": []})
-                    _tree = _tree[-1]["children"]
-
-        files: list[Path] = []
-        for result in expand_braces(self.glob):
-            files += list(Path(self.root_dir).resolve().glob(result))
-
-        files = [f for f in files if f != Path(self.root_dir).resolve()]
-
-        ignore_files = []
-        if self.ignore_glob:
-            for result in expand_braces(self.ignore_glob):
-                ignore_files += list(Path(self.root_dir).resolve().glob(result))
-            files = list(set(files) - set(ignore_files))
-
-        files_with_sep = []
-        for f in files:
-            file = str(f.relative_to(self.root_dir))
-            if f.is_dir():
-                file += os.path.sep
-            files_with_sep.append(file)
-
-        tree = make_tree(files_with_sep)
-        return tree
+        return folders + files
 
     def _safe_join(self, folders):
```

Comment:
Didn't notice this before, but we should use `routes.safe_join` instead of reimplementing this method

Reply:
gonna keep it separate for now, `routes.safe_join` returns HTTPExceptions that don't make sense in this context, and has a different signature.